DepotDownloader/ContentDownloader.cs

```diff
@@ -403,7 +403,7 @@ public static async Task DownloadAppAsync( uint appId, uint depotId, ulong manif
                 {
                     string contentName = GetAppOrDepotName( INVALID_DEPOT_ID, appId );
                     Console.WriteLine( "App {0} ({1}) is not available from this account.", appId, contentName );
-                    return;
+                    throw new Exception(String.Format("App {0} ({1}) is not available from this account.", appId, contentName));

```

Comment:
https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2201-do-not-raise-reserved-exception-types?view=vs-2019

Reply:
I changed the throws to a custom exception now, in order to catch them but not interfere with unexpected and thus interesting exceptions and their tracebacks.

My original idea was to just raise unhandled exceptions for unhandled errors, letting someone else change it to a more specific exception if they wanted to catch it at a later time (this would be bad in a library, but I don't expect anyone to reuse DepotDownloader in that way). Other parts of the code seem to use Exception() for this already.

On a related note, do you know if the linter that gives this warning is available with the dotnet cli tools? I don't have a VisualStudio setup.

Reply:
It is, we just haven’t configured it (yet).